src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonProperty.cs

```diff
@@ -25,8 +24,9 @@ internal JsonProperty(JsonElement value)
 
         /// <summary>
         ///   The name of this property.
+        ///   This allocates a new string instance for each call.
         /// </summary>
-        public string Name => _name ?? Value.GetPropertyName();
+        public string Name => Value.GetPropertyName();
```

Comment:
I'm guessing an `??=` was originally intended here. Why not fix it to that... To avoid that allocation you just publicly documented?

Reply:
It's because the type is a read-only struct. Changing this would require making it mutable.

Reply:
Thx. That makes sense!